File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/InstituteTeacherForm.ts
import { LookupEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface InstituteTeacherForm {
    InstituteId: LookupEditor;
    ClassId: LookupEditor;
    SemesterId: LookupEditor;
    SubjectId: LookupEditor;
    TeacherId: LookupEditor;
    InstituteDivisionId: LookupEditor;
    AcademicYearId: LookupEditor;
    Description: TextAreaEditor;
}

export class InstituteTeacherForm extends PrefixedContext {
    static readonly formKey = 'Institute.InstituteTeacher';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!InstituteTeacherForm.init)  {
            InstituteTeacherForm.init = true;

            var w0 = LookupEditor;
            var w1 = TextAreaEditor;

            initFormType(InstituteTeacherForm, [
                'InstituteId', w0,
                'ClassId', w0,
                'SemesterId', w0,
                'SubjectId', w0,
                'TeacherId', w0,
                'InstituteDivisionId', w0,
                'AcademicYearId', w0,
                'Description', w1
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Institute/InstituteTeacher/InstituteTeacherForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Institute.Forms;

[FormScript("Institute.InstituteTeacher")]
[BasedOnRow(typeof(InstituteTeacherRow), CheckNames = true)]
public class InstituteTeacherForm
{
    [HalfWidth]
    public int InstituteId { get; set; }
      
    [HalfWidth]
    public int ClassId { get; set; }
    [HalfWidth]
    public int SemesterId { get; set; }
   
    [HalfWidth]
    public int SubjectId { get; set; }
    [HalfWidth]
    public int TeacherId { get; set; }
    [HalfWidth]
    public int InstituteDivisionId { get; set; }

    [HalfWidth]
    public int AcademicYearId { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
   
    
}